### ISSUE TITLE
Improve output of trial TestCase assertions.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6306]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)| @tomprince reported|
|-|-|
|Trac ID|trac#6306|
|Type|enhancement|
|Created|2013-02-16 23:29:34Z|
|Branch|https://github.com/twisted/twisted/tree/better-assertion-messages-6306|

Python 2.7's `unittest.TestCase`'s `assertEqual`, `assertTrue`, and `assertFalse` (at least) provide nicer output than twisted's version of those. We should at least provide equivalent functionality when using 2.7.

<details><summary>Searchable metadata</summary>

```
trac-id__6306 6306
type__enhancement enhancement
reporter__tom_prince tom.prince
priority__normal normal
milestone__ 
branch__branches_better_assertion_messages_6306 branches/better-assertion-messages-6306
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1361057374000000 1361057374000000
changetime__1427887127584751 1427887127584751
version__None None
owner__exarkun exarkun
cc__jml
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="Julian's avatar" src="https://avatars.githubusercontent.com/u/329822?s=50" width="50" height="50">](https://github.com/Julian)<a name="note_2"></a>|@Julian commented|
|-|-|

I think the solution for this, along witih [#5338](https://github.com/twisted/twisted/issues/5338), [#5339](https://github.com/twisted/twisted/issues/5339) and possibly [#2690](https://github.com/twisted/twisted/issues/2690) is that trial should only define assertions on TestCase that are missing from TestCase.

Haven't filed a specific ticket for that though I've been meaning to implement it.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [44339]) Branching to 'better-assertion-messages-6306'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun removed owner|
|-|-|

I almost followed Julian's suggestion.  However, there are tests for the return values of these methods and the standard library's implementations don't pass them.  Therefore, the branch up for review leaves stubs in place that delegates the majority of the implementation to the standard library methods but adds the required return value behavior.  It may make sense to deprecate this behavior (somehow?) - I don't recall it ever being of any use - but that strikes me as a separate unit of work.

`assertEqual`, `assertTrue`, and `assertFalse` will now raise whatever failure the standard library raises.  Note that you have to set `longMessage` to `True` on your `TestCase` in order to get better output from `assertTrue` and `assertFalse` (which seems like nonsense to me - `assertEqual` will give you some extremely long output without setting a special flag, but oh well: `assertEqual` is the case that's most interesting here anyway).

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl set owner to @exarkun|
|-|-|

Hi exarkun!

Thanks for this!

- Pyflakes says that pformat isn't taken out of test_assertions as it was in _synctest.
- The newsfile only mentions assertEqual, but assertTrue and assertFalse's implementations have also changed. Might be worth adding those to the newsfile?
- The return value stuff is a bit icky, and I agree it should be deprecated. This is probably a terrible idea, but maybe by returning a proxy object of the condition, which emits DeprecationWarnings? However, that is a different unit of work, certainly.

Please fix [#11338](https://github.com/twisted/twisted/issues/11338), consider #2, file a ticket for #3 if you think there's a way of deprecating it, and merge with gusto. :)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [44345]) Merge better-assertion-messages-6306

Author: exarkun
Reviewer: hawkowl
Fixes: [#6306](https://github.com/twisted/twisted/issues/6306)

Change trial's `assertEqual`, `assertTrue`, and `assertFalse` methods to use
the corresponding standard library implementation and to pass through the more
informative failure message generated by those implementations on failure.
